libsolidity/ast/Types.cpp

```diff
@@ -1505,6 +1505,9 @@ TypeResult ReferenceType::unaryOperatorResult(Token _operator) const
 		return TypeProvider::emptyTuple();
 	case DataLocation::Storage:
 		return isPointer() ? nullptr : TypeProvider::emptyTuple();
+	case DataLocation::Code:
```

Comment:
Should cases such as these be added to `TypeProvider.cpp` as well, or do we already throw an error by the time we're supposed to reach `TypeProvider` functions?

Reply:
At the moment, yes, we should have an error at `SyntaxChecker` itself. But if required, I can add the skeletal code for `DataLocation::Code` as well. 

But it won't make any difference on the user end or any of the added test cases.